app/models/employer_events/renderer.rb

```diff
@@ -54,6 +54,55 @@ def renewal_and_no_future_plan_year?(carrier)
       !found_future_plan_year
     end
 
+    def find_plan_year_dates(carrier, date)
+      dates = carrier_plan_years(carrier).map do |node|
+      node.xpath("cv:#{date}", {:cv => XML_NS}).map do |date_node|
+         Date.strptime(date_node.content, "%Y%m%d") rescue nil
+        end
+      end    
+      dates.flatten.sort.last if dates.present? rescue nil
+    end
+    
```

Comment:
We'll have to work on how this fails and how we recover if the start date is either a missing or a blank node.  If I remember correctly the default sort gets super whiny if one of the elements in the sorted array is nada.

Also remember you want the begin and end dates to make the match - we want to weed out any cancelled plan years or anything else funky.  To this end we probably want to look up the start and end dates at the same time off the same node - it is unlikely but possible some weird things happen where we get an end date on the 'most recent' (according to the start date) node that is EARLIER than the 'most recent' (if you were to sort by end date) node.  What we want is the nodes for PY, sorted by start date, pick the last, and use the start and end date properties from that exact node.

Reply:
removing nils from the dates array

Reply:
adding guard statement in update_event_name method